tests/handlers/external_test.py

```diff
@@ -57,15 +63,17 @@ async def test_get_location(mocked_client: tuple[AsyncClient, RubinDataMocker])
 
 
 @pytest.mark.asyncio
+# @patch("lsst.ts.rubintv.main.HistoricalPoller")
 async def test_current_channels(
-    mocked_client: tuple[AsyncClient, RubinDataMocker]
+    mocked_client: tuple[AsyncClient, FastAPI, RubinDataMocker]
 ) -> None:
-    """Test that location page has links to cameras"""
-    client, mocker = mocked_client
-    print("in test_current_channels")
-    print(mocker.empty_channel)
-    # allow historicaldata time to sort info
-    await asyncio.sleep(1)
+    client, app, mocker = mocked_client
+
+    cp: CurrentPoller = app.state.current_poller
+    cp.test_mode = True
```

Comment:
The `app` comes from the `conftest.py` right? Maybe it would be better to set this test mode in the conftest so this is shared across all tests and you don't need to be setting it individually on specific ones (with the potential situation of missing a test). I think you could do that on: https://github.com/lsst-ts/rubintv/blob/edc06732e182a7a90458ca562735f84fe071053b/tests/conftest.py#L42
Something like:
```
...
app = create_app()
app.state.current_poller.test_mode = True
...
```

Reply:
It's not strictly necessary for the tests in `external.py` and at the moment `test_mode` means that the CurrentPoller only polls once before quitting which isn't something that every test will require. 
I'd rather refactor it later after writing the whole CurrentPoller test file.